### PR TITLE
Storage Content Transfer : BoH long-range dump, BPRED tweaks

### PR DESCRIPTION
**Pull request by LaharlMontogmmery**
Hello, station!

Lately, the Bluespace Rapid Part Exchanger Device (Also know as BREPD) has been the talk of our NT research departement. A great feat of engineering, being able to switch parts in a machine remotely! 

So we asked our research interns...I mean, experts to see how they could use the technology with other NT bluespace devices.

And this is how our brand new Bag of Holding was born! Fit with a new translocation technology, it can dump the content present in his pocket dimension anywhere, anytime! Here is some example for it's use :   

-Quick deliveries! May it be for emergency supplies, mining supplies...Any kind of supplies. Not BoHs, tho. Remember what happens when you fit a BoH in a BoH?
-Too lazy to reach that bin? No problem! (Works also if you want to dump the content of your BRPED in a bin. No more spare useless parts!)
-Offensive, tactical usages, for your local sec team! Ho dear, a cult transformed a part of their station into a impenetrable fortress, and the chaplain is not in sight? Ho dear me, the aliens in the xenobio pen went lose again? Give them a little gift, courtesy of NT Bluespace Technologies.

Disclaimer : NT is not responsible for any possible harm or reality alteration due to the use of this item. Do not let this item fall into the hands of the Syndicate : who knows what they might do with it.

---

So yes, the BoH now ignores the usual adjacency check (you still to have it near you, tho!) and can transfer/dump his content anywhere, like the BRPED. The work done in the previous PR allowed me to quickly implement this change, thankfully.
Comes with the PSHOOM sound and the laser thing of the BRPED, since they both use Bluespace tech.
BPRED has been nerfed for consistency, and also I don't think you're supposed to upgrade machines through cameras. That's silly.

### CODE REVIEW

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 1]
```diff
@@ -34,6 +34,8 @@
 	icon_state = "holdingpack"
 	max_w_class = 5
 	max_combined_w_class = 35
+	var/pshoom = 'sound/items/PSHOOM.ogg'
+	var/alt_sound = 'sound/items/PSHOOM_2.ogg'
```
Comment (Jordie0608) [inline code/game/objects/items/weapons/storage/backpack.dm]: These don't really need to be vars, just the path in playsound() works; unless you explicitly want both to be vareditable

**Comment by Jordie0608** [comment]
Changelog

**Comment by LaharlMontogmmery** [comment]
Yeah, added the changelog for my last change too since I forgot that.
Also, I'd like to keep them vareditable for now, unless it's really a problem.

**Comment by AlexCrimson** [comment]
So much power creep lately. Jesus fuck...

**Comment by Miauw62** [comment]
Rip malf AI
Op 21-jun.-2015 16:23 schreef "AlexCrimson" <notifications@github.com>:

**Comment by LaharlMontogmmery** [comment]
Deploying the nerf bat.

**Comment by LaharlMontogmmery** [comment]
Yeah, I'd prefer that people would not use the BoH through cameras, so I'm adding a distance cap to the translocation.

Another limit to that system is that you need to have vision of the turf in order to use the translocation power.

I'm currently reducing the range of the translocation to the vision range (which is 7), but I can reduce it further if you feel like it is necessary. NT'll blame it on the interns.

**Comment by Jordie0608** [comment]
Vareditable sounds aren't a problem, just wanted to be check that you meant to do it.

**Comment by pudl** [comment]
:-1:
I already complained to Goof that the bluespace rped was OP, I don't want the bag itself following suit.

**Comment by Anonus** [comment]
Two of the bags can ruin the station harder than anything else and if that doesn't count as OP then what even is?

**Comment by pudl** [comment]
@Anonus At least doing that is bannable as non-antag, and it kills you. This item can be used as a non-antag, it doesn't kill you, etc.

**Comment by LaharlMontogmmery** [comment]
@pudl You've put the finger on the problem  yourself : RnD can't efficiently deliver equipment, mainly because most of the place they can deliver the equipment to is generally off-limits, or they can't deliver it safely.

The BRPED was a step in the right direction : we see a lot more stuff outside of RnD getting upgraded.  The BoH would solve the equipment delivery issue, by circumvent the problem of windows and airlocks - which mean you could safely delivering equipment in a department.

**Comment by pudl** [comment]
@LaharlMontogmmery I didn't say that science has difficulty handing out equipment, I said that they don't. Any round where a considerate scientist puts devices on the table to the hallway is a round full of science stuff. There's no issue with transportation whatsoever.
The BRPED has proven to be useful, but it still keeps scientists in their department, and that's something I don't like. Scientists are like handymen who go around the station to upgrade stuff. Making them an invisible force of upgrading that just sit on a pile of guns and mechs is the wrong direction.

**Comment by LaharlMontogmmery** [comment]
@pudl Funnily, I think I've seen more scientists going out of their department upgrading Medbay ), mainly due to the fact that you don't need to screwdriver the machines anymore. Also the fact that goof allowed for more machines to be upgradable. Might be just me, tho.

Also, RnD can't exactly use the guns they're sitting on, due to firing pings. (It's funny how this actually avoid a Cargonia V2 situation, kind of.) So if they make guns, they have to bring them to the brig anyway.

But yes, Science is definitly not putting items on the table - but while it is true that RnD is not going out giving out equipement, I do generally see anyone coming to the front of RnD getting what they requested, if it's available.

 But I do see what you mean by "RnD holes up in his department." . But part of this is due to the fact that BRPED usage goes through cams - I just nerfed the BoH so it doesn't, to force the scientists to actually go out of their dept. It might be a good idea to also nerf the BRPED this way.

To finish, it would be nice to have some use for BoH, instead of just being a fancy bag that can potentially be the round ender that it is currently.

**Comment by pudl** [comment]
You know what? Nerf the BPRED like you said for some consistency, and I support this change. There's still a power creep, but as long as it stops here then I don't mind.

**Comment by LaharlMontogmmery** [comment]
That we can agree on.

I'll nerf the BRPED to have an actual range, to avoid future abuses - and for the sake of consistency.

Also, last opinion : this is about the range of the BoH/BRPED range of the translocation. I seriously want some opinions on this, so : 

Range 7(your vision range), Range 6, Range 5 or less?

**Comment by pudl** [comment]
7 doesn't hurt.

**Comment by LaharlMontogmmery** [comment]
Goodbye, camera upgrading.

We'll miss you.

**Comment by GunHog** [comment]
I am not willing to give up BSRPED remote upgrading for adding any such thing to the BoH. This will only be used for easy bombing of things. Science now has the ability to ship things via disposals by making a destination tagger, wrapping the item, and flushing it.

Things are *finally* getting upgraded by Science. The department is finally serving its purpose of improving other departments. To try and take such a good thing away is nothing short of awful, in my opinion. Massive :-1: on the BSRPED nerf. A lot of good will be lost with it.

**Comment by Miauw62** [comment]
Science staying in their department and having no reason to leave is dumb and bad design. If science wants to upgrade places, they should go there. 7 tile range makes upgrading machines much easier already, since all you need is view, so you can upgrade genetics without entering genetics still.

**Comment by LaharlMontogmmery** [comment]
Mind you, the range 7 is your FULL vision. You can upgrade a bunch of machinery just from the counters already - example, Genetics, and Botany.

I don't think upgrading stuff through cameras was ever intended.

**Comment by pudl** [comment]
Camera upgrading wasn't intended, Goof coded the feature, discovered the bug while playtesting, and decided that it shouldn't be fixed.

**Comment by Lumipharon** [comment]
Stop the R&D power creep.

**Comment by Iamgoofball** [comment]
@LaharlMontogmmery Upgrading through cameras is intentional. Please revert the BSRPED nerfs.

File: code/game/objects/items/weapons/storage/backpack.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/game/objects/items/weapons/storage/backpack.dm]: I set them as vars with the BSRPED implementation for var editing purposes.

**Comment by LaharlMontogmmery** [comment]
@Iamgoofball Camera upgrading is the only nerf to BREPD. That's it. Anyone can still upgrade any machinery in sight.

**Comment by Iamgoofball** [comment]
Alrighty.

**Comment by Lumipharon** [comment]
Goof, as far as I understand this PR (correct me if I'm wrong), is that with a BoH I can now teleport grenades/bombs/dildos anywhere on screen I can see (xray helps).

The brped already avoids almost any interaction with the crew since you just run down the hall and click all the machines. The extra capacity is good, not having to screwdriver is good. Long range interaction is, in my opinion, not good.

Also to clarify, will this let you Boh teleport things INTO other containers (such as another BoH)? That would be bad.

**Comment by optimumtact** [comment]
PSHOOM, remember people voted to make the game more lethal

**Comment by Iamgoofball** [comment]
@Lumipharon 
i don't see a problem with that
what is bad about scientists upgrading things more efficiently?

**Comment by Lumipharon** [comment]
1: Upgrades are so powerful, and now effect so many things, that most default machines are a joke compared to their upgraded state,

2: Going into departments to upgrade their shit means actual human interaction, as opposed to just remotely clicking on machines as you walk down the corridor.

**Comment by Iamgoofball** [comment]
@Lumipharon 
1. And?
2. Considering this wasn't happening I doubt anyone actually enjoys it. Don't fuck with the people who do things differently then you. If you want more player interaction then do it yourself and let people like GUN_HOG who just wants to upgrade shit and get back to work be.

**Comment by Lumipharon** [comment]
And being able to make 700+ sheets of minerals from ONE sheet of metal, or quadrupling mining output is not good for balance.

But it was happening. Of course it's easier (an thus happens more often) with the brped, but that doesn't make it good for gameplay.

**Comment by Iamgoofball** [comment]
@Lumipharon 
Are people having fun?
If so, it is good gameplay.

**Comment by Lumipharon** [comment]
The just make clicking the brped autoupgrade every machine on station, starting from what's closest to you.

I believe that more human interaction is good. If you disagree with that, then that's ok, I can't argue against that.

More on topic, can this be used to remotely singulo bomb or not?

**Comment by Iamgoofball** [comment]
@Lumipharon No, because you can't remotely put a BoH into another BoH.

**Comment by GunHog** [comment]
@Lumipharon My point is that human interaction was not occurring before. It simply did not happen. Tell me, how many Scientists walked all the way to Engineering to upgrade the SMES? If we had logs for it, I would bet I could count them on my hands. How about the heaters in atmos? Cargo's autolathe?

As RD, my experiences are thus, before the BSRPED:
- Robotics is usually the second to be upgraded after R&D.
- I regularly upgrade Medbay without being asked, the doctors generally thank me for it.
- The chef, on rare occasion, asks for microwave upgrades.
- I have once given the CE an RPED full of capacitors and cells for the SMES.
- Cargo will sometimes ask for parts to upgrade their autolathe.
- Rarely, an assistant will ask for parts.
- I have never upgraded anything in Botany.
- I have never upgraded anything in Atmospherics, or the turbine.

For you to say that this will increase interaction, in my opinion, is wholly inaccurate. This is a player psychology issue. Players are lazy, and they will not go out of their way to do anything that helps another, yet not themselves, unless it is easy to do. These interactions *rarely, if ever, happened*. I would go so far to say that the BSRPED update increased my interaction, just not in person. Since then, I have been asked for upgrades more often from Cargo, the chef, and even Atmos. The amount of interaction will be decreased in this nerf (as it is an annoyance to have to travel to another department to do an upgrade), but will have more than the RPED alone. Note that not all machines to be upgraded can be seen from the exterior. This is especially true for Engineering.

TL;DR, the BSRPED has certainly increased my fun, as well as everyone else I have remotely upgraded. This nerf will take that away. It will still be more fun than standard RPED times, but certainly not as much, nor will as many players benefit as before.

**Comment by Miauw62** [comment]
and this still makes it much easier for scientists to upgrade shit because you dont even need to enter the department most of the time, but it does prevent scientists from not having a reason to leave their department EVER, literally.

**Comment by palpatine213** [comment]
Rip upgrading the emitters. I'm pretty sure no scientist is going to go walk to the far emitters while the singulo is running.

**Comment by GunHog** [comment]
> and this still makes it much easier for scientists to upgrade shit because you dont even need to enter the department most of the time, but it does prevent scientists from not having a reason to leave their department EVER, literally.

Everything in Science, including robotics, is designed around progression and timesinks (except toxins). Due to this mechanic, and the general state of being a busy department, leaving Science means that whatever job that scientist was doing will suffer. I experience this all the time as RD. If I am handling my department's affairs, something will come up (almost without fail), and I must leave to deal with that, leaving people waiting.

Unless someone in the department is free of a duty or has a cover, the department may suffer from a staff member leaving to do upgrades. Leaving is simply not something that is a good thing to do as a Science member, unless you are the RD or need to collect things. I do not see this as a bad mechanic, just a by-product of Sci's time-locked nature and the effort required to progress.

My point is that Science staff typically will not leave often, and asking them to go across the station to serve their purpose of helping other departments improve is simply too much. It is a player issue, one that cannot be hand-waved off by telling the players to deal with it. *Do not expect Science players to leave more often after this nerf*, we will simply be moving back about halfway to the days before the BSRPED, as our players do not seem to consider upgrades to be a sufficient reason to leave the department, excepting Medbay or direct prodding.

**Comment by CorruptComputer** [comment]
Make upgrading things an engineering job, Sci researches it, Eng builds it.

**Comment by Miauw62** [comment]
let me rephrase what you said: leaving is something that is *mechanically undesirable* to do. but the game is improved by people not sitting in their department forever, so we must implement changes that encourage players to do something that is mechanically undesirable, whether that be by necessity or making it desirable.

**Comment by CorruptComputer** [comment]
@Miauw62
Me or @GunHog

**Comment by Kromgar** [comment]
Frankly I think BRPED upgrade through cams should stay. 

I never bothered to upgrade SMES's before. It was too much of a hassle before BRPED to go all over upgrading everything.

I used to only upgrade Medbay, and SCI occasionally the chefs shit if they asked. 

Nerfing the BRPED will not increase people going to Engineering to upgrade their machines. Or cargo save for the Ore Smelter. People are self-serving unless it isn't that much of a hassle.

Occasionally with the BRPED i don't bother with a security console and just run around upgrading everything but if I do get the security console I work  my hardest to upgrade everything. Every machine on the station I will attempt to make it better.

Nerfing the BRPED will see upgrades for departments other than Sci and medbay to be reduced.

**Comment by Kromgar** [comment]
Wait... you could upgrade the emitters? Can you upgrade the shield gen too?

**Comment by Iamgoofball** [comment]
why do you think I declared it a feature

**Comment by palpatine213** [comment]
Nerfing the BRPED doesn't increase the incentive for science to go outside their department, it just decreases their incentive to go upgrade everything. The nerf doesn't really solve anything, it just makes the rest of the station suffer while science isn't really affected that much.

**Comment by pudl** [comment]
@palpatine213 Absolutely no one suffers if science refuses to upgrade anything, because nothing in the station is some shitty functionless thing when it isn't upgraded. Upgrading equipment is purely optional, and only just generally helpful, not super important. So if science refuses to upgrade anything because it's hard, it doesn't kill the station.

**Comment by freerealestate** [comment]
>nothing in the station is some shitty functionless thing when it isn't
upgraded

what about the cloner, or sleepers?

**Comment by palpatine213** [comment]
@pudl That's the thing. Scientists basically could care less about departments other than medbay and cargo in terms of outside upgrades. If the goal is to get scientists out and about, there should be a REASON for them to really want to upgrade those things. It's better that the station is getting upgrades, and scientists are just sitting on their pile of mechs and guns versus the station NOT getting upgrades, and scientists still just sitting on their pile.

**Comment by GunHog** [comment]
There will be no positive impact to any players following this nerf, so why do it? Changing something that clearly has changed affected departments for the better for the sake of "consistency" or attempting to force Science players to trot about the station to do something that does not really benefit or reward them is an action to be reconsidered.

Please remove the BSRPED changes in this PR. The change itself is earmarked to this; It was not even the original design of the PR, just something that was demanded to be tacked on.

Speaking on the original design itself, I do not know how much use it will see beyond the initial hype period. It seems easier to simply mail items to the intended department. (Or simply ask them to come and get things themselves)

**Comment by Iamgoofball** [comment]
@GunHog 
Give them to miners. They can just walk up, teleport the resources into your office, and go back.

**Comment by pudl** [comment]
@freerealestate The only way a cloner needs to be upgraded is when you find a husk, which is almost NEVER EVER. That hardly makes it functionless. Also, sleepers are basically universally viewed as mostly useless. Even at maximum upgrading, they're not as good as just asking the chemist.
@palpatine213 Maybe the reason they never go out to upgrade anything is because there's no benefit to them upgrading other departments. For example, emitters were specifically made upgradable to kill blobs. So the only time you need to upgrade them is when they're out to kill a blob, aka, outside of engineering.
@GunHog There is a positive change, and it's that it enforces RP in a server that's moving more toward a heavier RP environment. Speaking of "consistency", Paprika specifically made receiving blood from medbay as slow as hell just so they would have to sit in medbay and possibly talk to doctors more often. Making scientists talk to more people, THAT'S consistent.

**Comment by Iamgoofball** [comment]
@pudl 

point 1: cloners

you do realize they go slow as shit when not upgraded, right? like, really slow?

point 2: benefits of upgrading
Upgraded emitters also use less power, so a depowered engineering can keep singulo containment up for way longer than normal.

point 3:
gameplay >>>> RP

receiving blood being slow as shit is bad for gameplay because it isn't fun

**Comment by palpatine213** [comment]
@pudl You do realize that no one actually talks to doctors while giving blood, right? Hell, doctors just take blood from corpses or genetics monkeys now. I'm sure they hold great conversations.

**Comment by GunHog** [comment]
> Paprika specifically made receiving blood from medbay as slow as hell just so they would have to sit in medbay and possibly talk to doctors more often. Making scientists talk to more people, THAT'S consistent.

That is your reasoning? The fact that you refer to the speed in such a profane manner indicates that you are not happy with it either. And to further counter, I would like to point out that the talk between doctor and patient occurs at the point the doctor attaches the IV, and again when they return (the good doctors will, at least) to check that the process is done. Making it take a long time does not really do much, as the doctor will be doing other things in that time.

Also, talking is not lost with the remote feature. To ensure you recieve your upgrade, you should contact the Science department and request it. Thank them afterwards. I do not know what your experiences seem to be, but the majority of mine, using a standard RPED, is receiving a request, proceeding to the department, waiting to be let in, doing the job, being thanked, and leaving. The portion of interaction lost in the BSRPED would be the 'waiting to be let in' portion. Is this what you consider valuable here? Oh, and scientists will certainly not be interacting with Engineering much after this nerf, unless they come to Science and get their own RPED.

**Comment by LaharlMontogmmery** [comment]
Alright, so I'm reverting the BPRED change, because  :

- It's a terrible to mix adding features, and then nerfing something alongside. It's a terrible deal (akin to taking a feature hostage), and it's something I'd like to avoid - and I'd rather keep the two separate.
-The BPRED change is apparently something that is his own debacle entirely, so if there's going to be a nerf PR, It would be preferable to put it in his own PR altogether.

**Comment by Iamgoofball** [comment]
Now that the BSRPED nerf is removed, :+1:

**Comment by GunHog** [comment]
Yeeeey! Thank you, Laharl :) :+1: from me now as well!

**Comment by LaharlMontogmmery** [comment]
Ha, crap, hold on a sec.
